src/explore-education-statistics-frontend/src/middleware/pages/__tests__/redirectPages.test.ts

```diff
@@ -0,0 +1,275 @@
+import _redirectService, {
+  Redirects,
+} from '@frontend/services/redirectService';
+import redirectPages from '@frontend/middleware/pages/redirectPages';
+import { NextResponse, NextRequest } from 'next/server';
+
+jest.mock('@frontend/services/redirectService');
+const redirectService = _redirectService as jest.Mocked<
+  typeof _redirectService
+>;
+
+describe('redirectPages', () => {
+  const redirectSpy = jest.spyOn(NextResponse, 'redirect');
+  const nextSpy = jest.spyOn(NextResponse, 'next');
+
+  const testRedirects: Redirects = {
+    methodologyRedirects: [
+      { fromSlug: 'original-slug-1', toSlug: 'updated-slug-1' },
+      { fromSlug: 'original-slug-2', toSlug: 'updated-slug-2' },
+    ],
+    publicationRedirects: [
+      { fromSlug: 'original-slug-3', toSlug: 'updated-slug-3' },
+      { fromSlug: 'original-slug-4', toSlug: 'updated-slug-4' },
+    ],
+  };
+
+  afterEach(() => {
+    redirectSpy.mockReset();
+  });
+
+  test('does not re-request the list of redirects once it has been fetched', async () => {
+    redirectService.list.mockResolvedValue(testRedirects);
+    const req = new NextRequest(
+      new Request('https://my-env/methodology/original-slug'),
+      {},
+    );
+    await redirectPages(req);
+
+    expect(redirectService.list).toHaveBeenCalledTimes(1);
+
+    const req2 = new NextRequest(
+      new Request('https://my-env/methodology/another-slug'),
+      {},
+    );
+    await redirectPages(req2);
+
+    expect(redirectService.list).toHaveBeenCalledTimes(1);
+  });
+
+  describe('redirect methodology pages', () => {
+    test('redirects the request when the slug for the requested page has changed', async () => {
+      redirectService.list.mockResolvedValue(testRedirects);
+      const req = new NextRequest(
+        new Request('https://my-env/methodology/original-slug-1'),
+        {},
+      );
+      await redirectPages(req);
+
+      expect(redirectSpy).toHaveBeenCalledTimes(1);
+      expect(redirectSpy).toHaveBeenCalledWith(
+        new URL('/methodology/updated-slug-1', req.url),
```

Comment:
To make this more explicit and clearer, could we use a fully qualified URL instead of the `req.url` as a base? i.e.

```ts
new URL('https://my-env/methodology/updated-slug-1')
```

Same for other similar cases in this test suite